service/src/account_funding.rs

```diff
@@ -0,0 +1,167 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use crate::error::{Error, ServiceResult};
+use codec::Encode;
+use itp_node_api_extensions::AccountApi;
+use itp_settings::worker::{
+	EXISTENTIAL_DEPOSIT_FACTOR_FOR_INIT_FUNDS, REGISTERING_FEE_FACTOR_FOR_INIT_FUNDS,
+};
+use log::*;
+use sp_core::{
+	crypto::{AccountId32, Ss58Codec},
+	sr25519, Pair,
+};
+use sp_keyring::AccountKeyring;
+use substrate_api_client::{rpc::WsRpcClient, Api, Balance, GenericAddress, XtStatus};
+
+/// Draft for an enclave wallet trait (WIP).
+pub trait EnclaveWalletInfo {
+	fn free_balance(&self) -> ServiceResult<Balance>;
+}
```

Comment:
Extracted the account funding functions out of `main.rs` into this separate module and added a trait to query the free balance of the enclave account (which is used by the metrics server now).
Maybe `EnclaveWalletInfo` is not the best name? Since the wallet as we propose it in #636 has different requirements and serves a different purpose than this trait here.

Reply:
I'd say `EnclaveWalletInfo` should be renamed to `EnclaveAccountInfo`. The chain itself does not have a notion of wallets.

Reply:
Yes I agree, that would be a better name 👍